include/seqan3/alignment/pairwise/policy/simd_affine_gap_policy.hpp

```diff
@@ -0,0 +1,179 @@
+ *          configuration.
+ */
+template <typename alignment_algorithm_t, simd_concept score_t, typename align_local_t = std::false_type>
+class simd_affine_gap_policy
+{
+private:
+    //!\brief Befriends the derived class to grant it access to the private members.
+    friend alignment_algorithm_t;
+
+    //!\brief The state of the alignment algorithm for affine gaps.
+    using alignment_state_t = alignment_algorithm_state<score_t>;
+
+    /*!\name Constructors, destructor and assignment
+     * \brief Defaulted all standard constructor.
+     * \{
+     */
+    constexpr simd_affine_gap_policy() noexcept = default;                                      //!< Defaulted
+    constexpr simd_affine_gap_policy(simd_affine_gap_policy const &) noexcept = default;             //!< Defaulted
+    constexpr simd_affine_gap_policy(simd_affine_gap_policy &&) noexcept = default;                  //!< Defaulted
+    constexpr simd_affine_gap_policy & operator=(simd_affine_gap_policy const &) noexcept = default; //!< Defaulted
+    constexpr simd_affine_gap_policy & operator=(simd_affine_gap_policy &&) noexcept = default;      //!< Defaulted
+    ~simd_affine_gap_policy() noexcept = default;                                               //!< Defaulted
+    //!\}
+
+    /*!\brief Computes the score of the current simd cell.
+     * \tparam cell_t The type of the current cell [for detailed information on the type see below].
+     * \param[in,out] current_cell The current cell in the dynamic programming matrix.
+     * \param[in,out] state        The state storing hot helper variables.
+     * \param[in]     score        The score of comparing the respective letters of the first and the second sequence.
+     *
+     * \details
+     *
+     * `cell_t` is the result type of dereferencing the zipped iterator over the respective alignment score matrix and
+     * the alignment trace matrix used inside of the seqan3::detail::alignment_matrix_policy. The first parameter
+     * stored in the zipped tuple is the seqan3::detail::alignment_score_matrix_proxy and the second value is the
+     * seqan3::detail::alignment_trace_matrix_proxy.
+     */
+    template <typename cell_t>
+    constexpr void compute_cell(cell_t && current_cell,
+                                alignment_algorithm_state<score_t> & state,
+                                score_t const score) const noexcept
+    {
+        // score_cell = seqan3::detail::alignment_score_matrix_proxy
+        // trace_cell = seqan3::detail::alignment_trace_matrix_proxy
+        auto & [score_cell, trace_cell] = current_cell;
+        constexpr bool with_trace = !decays_to_ignore_v<std::remove_reference_t<decltype(trace_cell.current)>>;
+        // Precompute the diagonal score.
+        score_t tmp = score_cell.diagonal + score;
+
+        if constexpr (with_trace)
+        {
+            auto mask = tmp < score_cell.up;
+            tmp = (mask) ? score_cell.up : tmp;
```

Comment:
This is essentially `tmp = max{score_cell.up, tmp}`. Do you think it makes sense to document these lines in this way so that it's easier to follow the calculations?

Reply:
I added a description to the function details block

Reply:
Yes, that's good!